src/plugins/network.ts

```diff
@@ -49,11 +49,25 @@ type RequestWithEntry = Request & {
 
 export class NetworkManager {
   private _browser: BrowserInfo;
+  private _barrierPromises = new Set<Promise<void>>();
   results: Array<NetworkInfo> = [];
   _currentStep: Partial<Step> = null;
 
   constructor(private driver: Driver) {}
 
+  private _addBarrier(page: Page, promise: Promise<void>) {
```

Comment:
Are we trying to ensure that we collect header information before the page closes, or just prevent the unhandled exceptions in this case, or both?

Reading through the logic, it's clear to me how this would prevent unhandled exceptions from the page closing before the main promise promise is fulfilled. I'm having more trouble understanding the logic for preventing the page from closing before the data is collected, if that's part of this.

Reply:
Any async information that we are extracting from the response object or request object should happen during the page lifecycle, 

if the page is closed, the associated information is lost and this barrier enforces that we capture the information on the network entry only during the existence of the page.